stdlib/public/core/CMakeLists.txt

```diff
@@ -179,7 +181,13 @@ else()
   #set(LINK_FLAGS
   #  -Wl,--whole-archive swiftRuntime -Wl,--no-whole-archive)
   if("${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
-    list(APPEND swift_core_private_link_libraries
+    if("${ICU_UC}" STREQUAL "")
+      message(FATAL_ERROR "ICU_UC undefined")
+    endif()
+    if("${ICU_I18N}" STREQUAL "")
+      message(FATAL_ERROR "ICU_I18N undefined")
+    endif()
```

Comment:
Use the `precondition` helper that we have.  It will simplify this:

    precondition(ICU_UC "ICU_UC must be defined")
    precondition(ICU_I18N "ICU_I18N must be defined")